contracts/MechMarketplace.sol

```diff
@@ -643,22 +769,21 @@ contract MechMarketplace is IErrorsMarketplace {
         // Increase the amount of mech service multisig delivered requests
         mapMechServiceDeliveryCounts[mechServiceMultisig]++;
 
+        // Increase mech requester karma
+        IKarma(karma).changeRequesterMechKarma(requester, msg.sender, int256(numRequests));
         // Increase mech karma that delivers the request
-        IKarma(karma).changeMechKarma(msg.sender, int256(requestDatas.length));
+        IKarma(karma).changeMechKarma(msg.sender, int256(numRequests));
 
         // Get balance tracker address
         address balanceTracker = mapPaymentTypeBalanceTrackers[IMech(msg.sender).paymentType()];
 
-        // Process payment
-        IBalanceTracker(balanceTracker).adjustRequesterMechBalances{value: msg.value}(msg.sender, requester,
```

Comment:
This was wrong - no additional payment can be transferred. Also, total delivery rate calculation happens in the balance tracker side.